frontend/src/components/AbstractModal.tsx

```diff
@@ -0,0 +1,66 @@
+import { Dialog, DialogContent, DialogTitle, DialogActions, Typography, makeStyles, IconButton } from '@material-ui/core';
+import CloseIcon from '@material-ui/icons/Close';
+import * as React from 'react';
+import Button from './Button';
+
+
+export type modalType =  {
+    title: string;
+    description: string; 
+    confirmText: string;
+    cancelText: string;
+    isOpen: boolean;
+    handleConfirm: (e) => void;
+    handleCancel: (e) => void;
+};
+
+const useStyles = makeStyles(theme =>({
+    modal:{
+        position:'absolute',
+        top: theme.spacing(5),
+        padding: theme.spacing(2)
+    },
+    modalContent:{
+        textAlign: 'center'
+    },
+    modalAction:{
+        justifyContent: 'center'
+    },
+    modalTitle: {
+        position: 'absolute',
+        top: theme.spacing(2),
+        right: theme.spacing(0.5)
+    }
+}))
+
+
+export default function AbstractModal(props) {
+    const [modal, setModal] = React.useState<modalType>(props);
+    const classes = useStyles()
+
+    React.useEffect(() => {
+        setModal(props);
+      }, [props]);
```

Comment:
By passing props to `useState`, you've set the default value. I don't believe you need to call `useEffect` to set it again. Although passing a prop and then using it in state is very weird. I would have expected one or the other, but not both.

Reply:
I've changed the the implementation by using the props like an object now.